### DISCUSSION TITLE
How would I make the opponent play an animation when the player character misses a note?

### DISCUSSION THREAD

**Discussion by macohi** [FunkinCrew/Funkin#3864] [Q&A] [answered]
In an update to a Mod I am working on I would like to make my opponent character laugh at Boyfriend but I don't know much on the scripting side of VSlice and mostly copied from the Base Game's Scripts in a previous mod.

Does anyone have a way of making the opponent character play an animation when the player misses a note?

**Comment by AbnormalPoof** [comment, marked as answer]
Funkin' has a lot of built-in callbacks for various things. This example uses `onNoteMiss`, which is called when you... well...
```haxe
function onNoteMiss(event:NoteScriptEvent)
{
	super.onNoteMiss(event);
	// Your custom functionality here
	// i.e: PlayState.instance.currentStage.getDad().playAnimation('animation-name');
	// If you want it to be forced: PlayState.instance.currentStage.getDad().playAnimation('animation-name', true, true);
}
```
NOTE: This will not run if you "miss" a note by mashing, there is a different callback for that (`onNoteGhostMiss`)

**Reply by macohi** [reply]
isn't that for if it's a player character

**Reply by moxie-coder** [reply]
yeah, that's what you wanted isn't it?

**Reply by macohi** [reply]
the character I want to play an animation is an opponent character, he is not going to be playable

**Reply by macohi** [reply]
oh wait no nvm I see what is going on

**Reply by macohi** [reply]
sorry

**Reply by EliteMasterEric** [reply]
`onNoteHit` and `onNoteMiss` get called when a note gets hit or missed by either side! Check the event's note data to see what side the note is on before continuing.

**Reply by macohi** [reply]
I can make the opponent miss a note?

**Reply by KoloInDaCrib** [reply]
> I can make the opponent miss a note?

Yeah you could do that, the opponent pico in week 3 pico mixes misses the notes when shot (the code for it is in phillyTrainErect stage's .hxc file)

**Reply by macohi** [reply]
thanks guys!
